alfasim_sdk/hook_specs.py

```diff
@@ -146,6 +146,18 @@ def update_plugins_secondary_variables(ctx: "void*") -> "int":
     """
 
 
+def update_plugins_secondary_variables_extra(ctx: "void*") -> "int":
+    """
+    Internal simulator hook to update plugin's secondary variables in the ExtraVarSolver scope.
```

Comment:
Just a opinion, you can used it or not 😋 
```
Hook used internally by the simulator to update the plugins secondary variables on the ExtraVarSolver scope.

This hook will be executed right after the routine ExtraVarSolver has being finished, as the last step before continuing with the normal flow.

Reply:
The second line is not what happens.